tests/scripts/hpo/hpo_get_plot_tests.sh

```diff
@@ -0,0 +1,344 @@
+	echo "******************************************"
+	echo "$response}"
+	echo "******************************************"
+	echo "${response}" > ${result_log}
+}
+
+
+# validate obtaining plot from HPO /plot API for invalid queries
+# input: test name 
+function get_plot_invalid_tests() {
+	__test_name__=$1
+
+	SERV_LOG="${TEST_DIR}/service.log"
+
+	# Deploy hpo
+	if [ ${cluster_type} == "native" ]; then
+		deploy_hpo ${cluster_type} ${SERV_LOG}
+	else
+		deploy_hpo ${cluster_type} ${HPO_CONTAINER_IMAGE} ${SERV_LOG}
+	fi
+	
+	# Check if HPO services are started
+	check_server_status "${SERV_LOG}"
+
+	IFS=' ' read -r -a get_plot_invalid_tests <<<  ${hpo_get_plot_tests[$FUNCNAME]}
+	for exp_trial in "${get_plot_invalid_tests[@]}"
+	do
+		# Get the length of the service log before the test
+		log_length_before_test=$(cat ${SERV_LOG} | wc -l)
+
+		TESTS_="${TEST_DIR}/${exp_trial}"
+		mkdir -p ${TESTS_}
+		LOG_="${TEST_DIR}/${exp_trial}.log"
+		result_log="${TESTS_}/${exp_trial}_result.log"
+
+		TEST_SERV_LOG="${TESTS_}/service.log"
+
+		echo "************************************* ${exp_trial} Test ****************************************" | tee -a ${LOG_} ${LOG}
+
+		# Get the experiment id from search space JSON
+		exp="valid-experiment"
+		current_id=$(echo ${hpo_post_experiment_json[${exp}]} | jq '.search_space.experiment_id')
+		current_name=$(echo ${hpo_post_experiment_json[${exp}]} | jq '.search_space.experiment_name')
+
+		# Post a valid experiment to HPO /experiment_trials API.
+		post_experiment_json "${hpo_post_experiment_json[$exp]}"
+
+		# Run the get plot API test
+		run_get_plot_test ${exp_trial}
+
+		# sleep for service logs to be updated
+		sleep 5
+
+		# Extract the lines from the service log after log_length_before_test
+		extract_lines=`expr ${log_length_before_test} + 1`
+		cat ${SERV_LOG} | tail -n +${extract_lines} > ${TEST_SERV_LOG}
+		
+		echo ""
+		echo "log_length_before_test ${log_length_before_test}"
+		echo "extract_lines ${extract_lines}"
+		echo ""
+
+		actual_result="${http_code}"
+
+		expected_result_="400"
+
+		expected_log_msg="${hpo_get_plot_msgs[$exp_trial]}"
+
+		echo "actual_result = $actual_result expected_result = $expected_result_"
+		compare_result ${exp_trial} ${expected_result_} "${expected_log_msg}" "${TEST_SERV_LOG}"
+		echo ""
+		
+		stop_experiment "$current_name"
+	done
+	
+	# Stop the HPO servers
+	echo "Terminating any running HPO servers..." | tee -a ${LOG_} ${LOG}
+	terminate_hpo ${cluster_type} | tee -a ${LOG_} ${LOG}
+	echo "Terminating any running HPO servers...Done" | tee -a ${LOG_} ${LOG}
+	
+	# Sleep for few seconds to reduce the ambiguity
+	sleep 5
+	echo "*********************************************************************************************************" | tee -a ${LOG_} ${LOG}
+}
+
+# Post a valid experiment to HPO /experiment_trials API, Query it using valid experiment id and trial number and validate the plot API
+# input: test name
+function get_plot_valid_tests() {
```

Comment:
I think this entire function needs to be broken up into smaller functions. The function that has the inner HPO loop can be made common and reused for all tests (by passing in the right benchmark etc). We need a separate function for only the plots. This can be used elsewhere as well